### PR TITLE
Improvements to SimpleHostRoutingFilter

### PR DESCRIPTION
**Pull request by liebowitz**
Hi - long time Spring devotee, first time committer. Here is a summary of the PR:

- Extracted superclass for HostRoutingFilter. Note, there is a breaking change in ZuulProxyConfiguration.
- Rewrote SimpleHostRoutingFilter to use HttpClientBuilder and thread-safe PoolingHttpClientConnectionManager. Removed deprecated API references.
- Exposed methods in SimpleHostRoutingFilter to instantiate HttpClient and ConnectionManager so it is possible to override the default client.
- Fixed an issue in ProxyRequestHelper where the ignoredHeaders are always empty when set outside of the request execution thread.

Thanks.

### CODE REVIEW

**Comment by spencergibb** [comment]
@liebowitz have you signed the [contributor's agreement](https://support.springsource.com/spring_committer_signup)?

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyConfiguration.java [hunk 1]
```diff
@@ -34,10 +34,7 @@
 import org.springframework.cloud.netflix.zuul.filters.ProxyRouteLocator;
 import org.springframework.cloud.netflix.zuul.filters.ZuulProperties;
 import org.springframework.cloud.netflix.zuul.filters.pre.PreDecorationFilter;
-import org.springframework.cloud.netflix.zuul.filters.route.RestClientRibbonCommandFactory;
-import org.springframework.cloud.netflix.zuul.filters.route.RibbonCommandFactory;
-import org.springframework.cloud.netflix.zuul.filters.route.RibbonRoutingFilter;
-import org.springframework.cloud.netflix.zuul.filters.route.SimpleHostRoutingFilter;
+import org.springframework.cloud.netflix.zuul.filters.route.*;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyConfiguration.java]: Don't use `*` imports please.

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java [hunk 2]
```diff
@@ -49,17 +40,12 @@
...
-	 */
-	public static final String IGNORED_HEADERS = "ignoredHeaders";
-
 	public static final String CONTENT_ENCODING = "Content-Encoding";
 
 	private TraceRepository traces;
+    private HashSet<String> ignoredHeaders = new HashSet<>();
 
-	public void setTraces(TraceRepository traces) {
+    public void setTraces(TraceRepository traces) {
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java]: We use tabs, not spaces.

**Comment by spencergibb** [comment]
Can you clean up the formatting? It makes the `SimpleHostRoutingFilter` diff almost impossible to read.

**Comment by spencergibb** [comment]
Why extract `HostRoutingFilter` as a superclass if only one class extends it?

**Comment by liebowitz** [comment]
Contributor agreement has been signed.
Will make cleanup changes and resubmit.
In hindsight, SimpleHostRoutingFilter can remain as a single class without HostRoutingFilter... The goal here was to make SimpleHostRoutingFilter extensible.

**Pull request closed without merging**